src/ComputeSharp.D2D1/Shaders/Interop/Extensions/ID2D1EffectContextExtensions.cs

```diff
@@ -63,4 +64,75 @@ public static unsafe int GetD2D1Multithread(this ref ID2D1EffectContext effectCo
 
         return hresult;
     }
+
+    /// <summary>
+    /// Checks whether the input shader is not supported by the target <see cref="ID2D1EffectContext"/> instance.
+    /// </summary>
+    /// <param name="effectContext">The input <see cref="ID2D1EffectContext"/> instance.</param>
+    /// <param name="bytecode">The shader bytecode.</param>
+    /// <param name="bytecodeSize">The size of <paramref name="bytecode"/>.</param>
+    /// <returns>Whether the shader is not supported.</returns>
+    /// <remarks>If any calls needed to check support fail, the method will also return <see langword="false"/>.</remarks>
+    public static bool IsShaderNotSupported(this ref ID2D1EffectContext effectContext, byte* bytecode, int bytecodeSize)
+    {
+        using ComPtr<ID3D11ShaderReflection> d3D11ShaderReflection = default;
+
+        // Create the reflection instance, and in case of error just return the previous error like above
```

Comment:
comment seems out of date -- you're returning `false`, not "the previous error like above"

Reply:
Fixed, thanks! 🙂